src/tests/system/tests/test_sssctl.py

```diff
@@ -242,287 +180,159 @@ def test_sssctl__check_missing_domain_name(client: Client):
+            "id_provider = invalid",
+            "Attribute 'id_provider' in section 'domain/local' has an invalid value: invalid",
+            marks=[pytest.mark.ticket(bz=2100789), pytest.mark.importance("high")],
+        ),
+        pytest.param(
+            "id_provider.*",
+            "",
+            "Attribute 'id_provider' is missing in section 'domain/local'.",
+            marks=[pytest.mark.ticket(bz=2100789), pytest.mark.importance("high")],
+        ),
+        ("id_provider", "id_@provider", "Attribute 'id_@provider' is not allowed in section"),
+        ("domain/", "d$main/", "Section [d$main/local] is not allowed"),
+        ("domain/local", "domain/local@", "Section [domain/local@] is not allowed"),
+        ("domain/local", "domainlocal", "Section [domainlocal] is not allowed"),
+        (".sssd.", "[sssdx]", "Section [sssdx] is not allowed"),
+        (".pam.", "[pamx]", "Section [pamx] is not allowed"),
+        (".nss.", "[nssx]", "Section [nssx] is not allowed"),
+    ],
+)
 @pytest.mark.tools
 @pytest.mark.topology(KnownTopology.Client)
-def test_sssctl__check_invalid_section_in_name_in_snippet(client: Client):
+def test_sssctl__check_invalid_semantic_in_section_name(client: Client, pattern: str, repl: str, expected: str):
     """
-    :title: sssctl config-check detects invalid domain name in snippet
+    :title: sssctl prints appropriate error message if semantic in section name is invalid
     :setup:
-        1. Create new conf snippet with invalid domain name
+        1. Start SSSD
+        2. Edit sssd.conf with invalid semantic in section name
     :steps:
-        1. Call sssctl config-check
-        2. Check error message
+        1. Validate configuration file using sssctl config-check
     :expectedresults:
-        1. config-check detects an error in config snippet
-        2. Error message is properly set
-    :customerscenario: False
-    """
-    client.sssd.common.local()
-    client.fs.write("/etc/sssd/conf.d/01_snippet.conf", "[invalid/local]\ninvalid_option = True", mode="600")
-
-    result = client.sssctl.config_check()
-    assert result.rc != 0, "Config-check did not detect misconfigured config snippet"
-    assert "Section [invalid/local] is not allowed" in result.stdout, "Wrong error message was returned"
-
-
-@pytest.mark.tools
-@pytest.mark.topology(KnownTopology.Client)
-def test_sssctl__check_missing_equal_sign(client: Client):
-    """
-    :title: sssctl config-check detects missing equals sign
-    :setup:
-        1. Start SSSD, so default config is automatically created
-        2. Edit config file so "=" is missing
-    :steps:
-        1. Call sssctl config-check
-        2. Check error message
-    :expectedresults:
-        1. config-check detects an error
-        2. Error messages are properly set
+        1. sssctl configuration check fails with the correct output in stdout
     :customerscenario: False
     """
     client.sssd.common.local()
     client.sssd.start()
```

Comment:
I still think no 'sssctl.config_check' tests need to start SSSD.

Reply:
Explicit apply-config is better (from "intention" point of view) and faster.

Reply:
I agree. But let's do it in the separate PR because it's not related with parametrization and all tests in this module should be updated.

Reply:
ok